app/experimenter/experiments/tests/test_models.py

```diff
@@ -2143,6 +2155,25 @@ def test_latest_rejection_returns_rejection_for_review_to_idle(self):
         self.assertEqual(experiment.changes.latest_review_request(), changes[0])
         self.assertEqual(experiment.changes.latest_rejection(), changes[1])
 
+    def test_latest_rejection_returns_rejection_for_review_to_dirty(self):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.LIVE_DIRTY,
+        )
+
+        changes = []
+        for publish_status in (
+            NimbusExperiment.PublishStatus.REVIEW,
+            NimbusExperiment.PublishStatus.DIRTY,
+        ):
+            experiment.publish_status = publish_status
+            experiment.save()
+            changes.append(
+                generate_nimbus_changelog(experiment, experiment.owner, "test message")
+            )
```

Comment:
Oh yeah I would definitely just make lifecycles for these states since that'll do what you're doing here automatically

Reply:
Same as above